src/main.rs

```diff
@@ -49,7 +49,13 @@ use alacritty::util::fmt::Red;
 fn main() {
     // Load command line options and config
     let options = cli::Options::load();
-    let config = load_config(&options);
+    let mut config = load_config(&options);
+
+    // Override configurations based on CLI options.
```

Comment:
This could be better, maybe `config.override_with_options(&options)` could help? Would allow this to avoid being a `mut` binding really nicely.

Reply:
I don't think there's anything wrong with making this mutable. If we want to mutate `config`, having it mutable is the right thing to do. `config.override_with_options` would still require `config` to be mutable, so that wouldn't help here.

Reply:
I could be wrong, but it *should* allow for the binding to be mutable, since the mutation would be happening before it's bound.

Reply:
Yeah, `let conf = load_config(&options).override(options);` wouldn't require `conf` to be mutable, however that's just because it's temporarily mutable and then dropped again. Simalar to this:
```
let conf = {
    let mut conf = load();
    conf.mutate();
    conf
}
```

I'd say if a variable needs to be mutable, just make it mutable. No reason to try and work around it, because after all that's exactly what the `mut` keyword is for.